src/LazySets.jl

```diff
@@ -25,6 +25,40 @@ import .Assertions: activate_assertions, deactivate_assertions
 # activate assertions by default
 activate_assertions(LazySets)
 
+# =================================================
+# Glue code for optional dependencies (Requires.jl)
+# =================================================
+
+"""
+    require(package::Symbol; fun_name::String="", explanation::String="")
+
+Helper method to check for optional packages and printing an error message.
+
+### Input
+
+- `package`     -- symbol of the package name
+- `fun_name`    -- (optional; default: `""`) name of the function that requires
+                   the package
+- `explanation` -- (optional; default: `""`) additional explanation in the error
+                   message
+
+### Output
+
+If the package is loaded, this function has no effect.
+Otherwise it prints an error message.
+
+### Algorithm
+
+This function uses `@assert` and hence loses its ability to print an error
+message if assertions are deactivated.
+"""
+function require(package::Symbol; fun_name::String="", explanation::String="")
```

Comment:
is this change related to the PR? also, this check can also be done with a macro, an implementation is https://github.com/mforets/ReachabilityAnalysis.jl/blob/master/src/Initialization/init.jl#L62

Reply:
In the end it was not needed but if we want to use `require` in other modules then this is necessary, so I kept it.

Reply:
do you mean other submodules of LazySets? can't you just import the function in submodules?

Reply:
I did, but the function must be exported before it can be imported. But we load `Approximations.jl` before the function was defined in `init.jl`.

https://github.com/JuliaReach/LazySets.jl/blob/894f77bd2309a80e02d5be50acfb6dbc7c617947/src/LazySets.jl#L137-L157

Reply:
> I did, but the function must be exported before it can be imported.

not sure that i understood. why do you want to `import`? but we can do `using LazySets: foo` inside the `Approximations` module as for other functions (eg. `_rtol`).

Reply:
By *imported* I meant `using` of course.

> but we can do using LazySets: foo inside the Approximations module

Yes, but only after moving the function upward.

Reply:
I removed this change now as it was not needed.